lib/client/client.js

```diff
@@ -685,11 +697,31 @@ class Client {
       this.wallet.options.rpcServerAddr = '';
     }
 
-    ws.onopen = () => {
-      ws.send(JSON.stringify({
-        Action: 'setClient',
+    this.challengeHandler = new Promise(async (resolve, reject) => {
```

Comment:
I don't think it's a good idea to set and use `this.challengeHandler` and `this.challengeDone`. When two reconnect occur at about the same time, the first `authChallenge` response can resolve the second `challengeHandler`. A better way is to use local variables they won't affect each other.